neo4j/src/main/java/org/hibernate/ogm/datastore/neo4j/HttpNeo4jDialect.java

```diff
@@ -486,6 +490,46 @@ public void forEachTuple(ModelConsumer consumer, TupleTypeContext tupleTypeConte
 		consumer.consume( tupleSupplier );
 	}
 
+	@Override
+	public ClosableIterator<Tuple> callStoredProcedure(String storedProcedureName,
+			ProcedureQueryParameters queryParameters, TupleContext tupleContext) {
+		Map.Entry<String, Map> queryAndParams = buildProcedureQueryWithParams( storedProcedureName, queryParameters );
+		Statement statement = new Statement( queryAndParams.getKey(), queryAndParams.getValue() );
+		statement.setResultDataContents( Collections.singletonList( Statement.AS_ROW ) );
+		Statements statements = new Statements();
+		statements.addStatement( statement );
+		Long txId = transactionId( tupleContext.getTransactionContext() );
+		StatementsResponse response;
+		if ( txId != null ) {
+			response = client.executeQueriesInOpenTransaction( txId, statements );
+		}
+		else {
+			response = client.executeQueriesInNewTransaction( statements );
```

Comment:
I'm just thinking about: if there isn't an ongoing transaction, should be allow to execute a procedure? It seems that neo4j doesn't allow that. Should we allow it or not?

Reply:
Operations with Neo4j should always occur inside a transactions, we wrote this in the [documentation](https://docs.jboss.org/hibernate/ogm/5.4/reference/en-US/html_single/#ogm-neo4j-transactions):

```
In Neo4j, operations must be executed inside a transaction.

Make sure your interactions with Hibernate OGM are within a transaction when you target Neo4J.
```

Make sure to check what we do in `executeBackendQuery` and reuse the same code we have when the returned entity is not specified.